libjps/include/jps.h

```diff
@@ -0,0 +1,101 @@
+#ifndef _JPS_H_
+#define _JPS_H_
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct {
+    void * sim_context;
+} SimulationContext;
+
+typedef struct {
+    bool has_error;
+    char * error_message;
+} Status;
+
+typedef struct {
+    double x;
+    double y;
+} Coordinate;
+
+typedef struct {
+    int id;
+    Coordinate coordinate;
+    double velocity;
+    double v_phi;
+} Agent;
+
+typedef struct {
+    Agent * agents;
+    size_t size;
+} Agents;
+
+typedef struct {
+    int * goals;
+    size_t num_golas;
+} Route;
+
+/**
+ * @name JPSlib Functions.
+ * @param [in]  simulation_context  The simulation context (owned by JPSlib).
+ * @param [out] status              Status object containing error information and message.
+ * @{
+ **/
+
+/**
+ * Initialize the simulation
+ * @param [in]  geometry_file The filepath to the geometry configuration file.
+ * @return                    The simulation context.
+ **/
+extern SimulationContext initialize(const char * geometry_file, Status * status);
+
+/**
+ * Adds a new agent to the pedestrian simulation.
+ *
+ * @param [in] agent_id           The unique identifier for the newly created agent.
+ * @param [in] starting_area      The unique identifier of the starting area to place the new agent.
+ * @param [in] route              The goals of the pedestrian reflecting the desired route.
```

Comment:
Doesn't that imply fixed routing? Or is this only exit points? What about route changes to dynamic events, e.g. doors, trains? Also who is allocating this memory? I think there is a creation method missing. Looking at the API it seems the library is taking over memory ownership (which is ok and convenient) hence you need to make sure that memory is allocated from some library call.

Reply:
I changed it to int return, -1 for pedestrian could not be added.